src/nupic/types/Fraction.cpp

```diff
@@ -366,7 +366,7 @@ namespace nupic
 
     while(components.size())
     {
-      std::swap<int>(numerator_, denominator_);
+      std::swap(numerator_, denominator_);
```

Comment:
> Why did you remove template param?

This was presenting problems on MSVC (20 errors). So I had to removed type when use `swap` on MSVC as explanned here: 

https://bobobobo.wordpress.com/2011/06/07/_alloc-must-be-a-class-or-namespace-when-followed-by/